### ISSUE TITLE
Dark mode switch break navigator

### ISSUE THREAD

**Issue by cl3m** [Tlaster/PreCompose#18]
When switching the dark mode on or off with the command:
```
adb shell "cmd uimode night no"
adb shell "cmd uimode night yes"
```
The navigator does not work any more on android (it does work on iOS). You can check the sample project here : https://github.com/cl3m/DarkModePreComposeBug
It does work as expected with navigation-compose :
https://github.com/cl3m/DarkModePreComposeBug/tree/dedfa1229e188ac5cc3355156d2a1b1045eab11e

Any idea why?

**Comment by Tlaster** [comment]
Thanks for the bug report.
The `Navigator` is not retained during configure changes, and `rememberNavigator` is just simply `return remember { Navigator() }`, a quick workaround is to ignore configure changes.
I will fix it later.

**Comment by Phaestion** [comment]
This also affects our app. When running in a resizable window the navigation/route graph is also cleared.

**Comment by Tlaster** [comment]
This should be fixed after 1.3.13, and since I change the library dependencies, the sample project requires adding dependencies compose.foundation and compose.animation in order to get it properly running.
